cvxpy/lin_ops/canon_backend.py

```diff
@@ -1103,6 +1090,305 @@ def _to_dense(x):
+        """
+        Multiply view with constant data from the left.
+        """
+        lhs, is_param_free_lhs = self.get_constant_data(lin.data, view, column=False)
+        if is_param_free_lhs:
+            reps = view.rows // lhs.shape[-1]
+            stacked_lhs = (sp.kron(sp.eye(reps, format="csc"), lhs))
+
+            def func(x, _p):
+                return stacked_lhs.tocsc() @ x
+        else:
+            reps = view.rows // next(iter(lhs.values())).shape[-1]
+            eye = sp.eye(reps, format="csc")
+
+            if reps > 1:
+                stacked_lhs = {}
+                for param_id, param_mat in lhs.items():
+                    inds = np.arange(param_mat.shape[0])
+                    sub_inds = np.split(inds, self.param_to_size[param_id])
+                    stacked_lhs[param_id] = sp.vstack([sp.kron(eye, param_mat[sub_ind],
+                                                               format='csc')
+                                                       for sub_ind in sub_inds], format='csc')
+            else:
+                stacked_lhs = lhs
+
+            def parametrized_mul(x):
+                return {k: v @ x for k, v in stacked_lhs.items()}
+
+            func = parametrized_mul
+        return view.accumulate_over_variables(func, is_param_free_function=is_param_free_lhs)
+
+    @staticmethod
+    def promote(lin: LinOp, view: StackedSlicesTensorView) -> StackedSlicesTensorView:
+        num_entries = int(np.prod(lin.shape))
+        rows = np.zeros(num_entries)
+        view.select_rows(rows)
+        return view
+
+    def mul_elem(self, lin: LinOp, view: StackedSlicesTensorView) -> StackedSlicesTensorView:
+        """
+        Given (A, b) in view and constant data d, return (A*d, b*d).
+        When dealing with parametrized constant data, we need to repeat the variable tensor p times
+        and stack them vertically to ensure shape compatibility for elementwise multiplication
+        with the parametrized expression.
+        """
+        lhs, is_param_free_lhs = self.get_constant_data(lin.data, view, column=True)
+        if is_param_free_lhs:
+            def func(x, _p):
+                return lhs.multiply(x)
+        else:
+            reps = view.rows // next(iter(lhs.values())).shape[-1]
+
+            def parametrized_mul(x):
+                new_x = sp.vstack([x] * reps)
+                return {k: v.multiply(new_x) for k, v in lhs.items()}
+
+            func = parametrized_mul
+        return view.accumulate_over_variables(func, is_param_free_function=is_param_free_lhs)
+
+    @staticmethod
+    def sum_entries(_lin: LinOp, view: StackedSlicesTensorView) -> StackedSlicesTensorView:
+        def func(x, p):
+            if p == 1:
+                return sp.csr_matrix(x.sum(axis=0))
+            else:
+                m = x.shape[0] // p
+                return (sp.kron(sp.eye(p, format="csr"), np.ones(m)) @ x).tocsr()
```

Comment:
Here we can simplify sum_entries by directly passing in np.ones(m) as it will already be a row vector.